app/javascript/packs/modal.js

```diff
@@ -0,0 +1,36 @@
+document.addEventListener('DOMContentLoaded', function(){
+  var buttons = document.querySelectorAll(".btn");
+  var closes = document.querySelectorAll(".modal-close");
+  
+  function openModal(event) {
+    var modal = document.querySelector("div[id="+event.currentTarget.id+"]");
+    modal.classList.add('opacity-100');
+    modal.classList.add('z-50');
+  }
+  
+  function closeModal(event) {
+    var modal = document.querySelector("div[id="+event.currentTarget.id+"]");
```

Comment:
you can use only `document.getElementById()` for this

Reply:
We use the query selector in this way, because we have two elements with the same id, which are mutually related: the link button and the modal.